### PR TITLE
[DT-1227] Client self update of games hub update get settings

### PR DESCRIPTION
**Pull request by ricardo-frango**
**What does this PR do?**

   Add new method to get jsons from FeatureFlags object.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] FeatureFlags.kt

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APP-1227](https://aptoide.atlassian.net/browse/DT-1227)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-1227](https://aptoide.atlassian.net/browse/DT-1227)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[APP-1227]: https://aptoide.atlassian.net/browse/APP-1227?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-1227]: https://aptoide.atlassian.net/browse/APP-1227?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: feature-flags/src/main/java/cm/aptoide/pt/feature_flags/domain/FeatureFlags.kt [hunk 1]
```diff
@@ -44,4 +45,8 @@ class FeatureFlags @Inject constructor(
       result.filterNot { it.isBlank() }
     } ?: emptyList()
   }
+
+  suspend fun <T> getObject(key: String, klass: Class<T>) = mutex.withLock {
+    runCatching { Gson().fromJson(key, klass) }.getOrNull()
```
Comment (G0retZ) [inline feature-flags/src/main/java/cm/aptoide/pt/feature_flags/domain/FeatureFlags.kt:50]: Does it expect an actual `JSON` string provided in a `key` parameter? 🤔

Comment (ricardo-frango) [reply feature-flags/src/main/java/cm/aptoide/pt/feature_flags/domain/FeatureFlags.kt]: fixed

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: extensions/src/main/java/cm/aptoide/pt/extensions/StringExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package cm.aptoide.pt.extensions
+
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+object StringExtensions {
```
Comment (G0retZ) [inline extensions/src/main/java/cm/aptoide/pt/extensions/StringExtensions.kt:7]: What for do we need this object wrapping the extension function? 🤔

File: extensions/src/main/java/cm/aptoide/pt/extensions/DateExtensions.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package cm.aptoide.pt.extensions
+
+import java.util.Date
+import java.util.Locale
+import android.icu.text.DateFormat
+
+object DateExtensions {
```
Comment (G0retZ) [inline extensions/src/main/java/cm/aptoide/pt/extensions/DateExtensions.kt:7]: What for do we need this object wrapping the extension function? 🤔

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: extensions/src/main/java/cm/aptoide/pt/extensions/StringExtensions.kt (hunk 2 above)
Comment (G0retZ) [inline extensions/src/main/java/cm/aptoide/pt/extensions/StringExtensions.kt:7]: What for do we need this object wrapping the extension function? 🤔

File: extensions/src/main/java/cm/aptoide/pt/extensions/DateExtensions.kt (hunk 3 above)
Comment (ricardo-frango) [reply extensions/src/main/java/cm/aptoide/pt/extensions/DateExtensions.kt]: my bad. fixed

File: extensions/src/main/java/cm/aptoide/pt/extensions/StringExtensions.kt (hunk 2 above)
Comment (ricardo-frango) [reply extensions/src/main/java/cm/aptoide/pt/extensions/StringExtensions.kt]: my bad. fixed

**Review by G0retZ — APPROVED** [review]

**Pull request merged**